### PR TITLE
fix: imageresizer gray

### LINKED ISSUE #165: ImageResizer doesn't work on gray images
**Describe the bug**
`ImageResizer` doesn't work on gray images

**Describe how you solve it**

- - -

**Environment**

**Screenshots**

┆Issue is synchronized with this [Jira Task](https://jinaai.atlassian.net/browse/JINAHUB-45) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by carlosb1**
it closes #165 

* added exception in the image resizer to work with 2d arrays or 3d arrays with shape (x, y, 1)

### CODE REVIEW

File: crafters/image/ImageResizer/tests/test_imageresizer.py [hunk 1]
```diff
@@ -23,3 +33,41 @@ def test_resize():
...
+
+    # Test for tuple/list target_size
+    output_dim = (img_height, img_width)
+    crafter = ImageResizer(target_size=output_dim)
+    img_array = create_random_gray_img_array(img_width, img_height)
+    crafted_doc = crafter.craft(img_array)
+    assert crafted_doc['blob'].shape == output_dim
+
+
+def test_resize_gray_2d():
```
Comment (JoanFM) [inline crafters/image/ImageResizer/tests/test_imageresizer.py:57]: can these 2 tests unified parametrizing the input image?

**Review by JoanFM — COMMENTED** [review]
Seems like an interesting feature to have in different loaders of the different hub image executors.

**Review by JoanFM — CHANGES_REQUESTED** [review]
please bump the version so that the build tests are executed

File: crafters/image/ImageResizer/tests/test_imageresizer.py (hunk 1 above)
Comment (carlosb1) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:57]: the asserts are different. I am not sure if it is possible.

Comment (JoanFM) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:57]: how are they different the asserts of `test_resize_gray_2d` and `test_resize_gray`?  I just see the difference in the method used to create the random arrays?

File: crafters/image/ImageResizer/tests/test_imageresizer.py [hunk 2]
```diff
@@ -52,22 +55,3 @@ def test_resize_gray():
     img_array = create_random_gray_img_array(img_width, img_height)
```
Comment (JoanFM) [inline crafters/image/ImageResizer/tests/test_imageresizer.py:55]: why do we need this second part of the test? can't it be tested on the previous crafted?

File: crafters/image/ImageResizer/tests/test_imageresizer.py [hunk 3]
```diff
@@ -52,6 +52,5 @@ def test_resize_gray(img_array):
     # Test for tuple/list target_size
     output_dim = (img_height, img_width)
     crafter = ImageResizer(target_size=output_dim)
-    img_array = create_random_gray_img_array(img_width, img_height)
     crafted_doc = crafter.craft(img_array)
```
Comment (JoanFM) [inline crafters/image/ImageResizer/tests/test_imageresizer.py:55]: no need to craft again right?

Comment (carlosb1) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:55]: > no need to craft again right?

I am not sure... I would try to follow the old test structure.. I don' have any particula opinion

Comment (JoanFM) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:55]: it should be stateless so same input should lead to same output

Comment (carlosb1) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:55]: I don't undertand your proposal.. do you want to modify the asserts or do you want to remove one craft operation?

Comment (JoanFM) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:55]: one craft operation

Comment (carlosb1) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:55]: We can remove it in both tests... It should be the same structure for both test

**Review by JoanFM — APPROVED** [review]

File: crafters/image/ImageResizer/tests/test_imageresizer.py (hunk 3 above)
Comment (JoanFM) [reply crafters/image/ImageResizer/tests/test_imageresizer.py:55]: Sorry, I was confused, output_dim has changed

**Comment by JoanFM** [comment]
Thank you very much for the great contribution @carlosb1 !

**Pull request merged**